### ISSUE TITLE
Lint/TopLevelReturnWithArgument confuses arguments with conditionals

### ISSUE THREAD

**Issue by tas50** [rubocop-hq/rubocop#8462]
## Expected behavior

Lint/TopLevelReturnWithArgument should detect arguments to return, but not conditionals.

## Actual behavior

Incorrectly detects return used with `if/unless`

## Steps to reproduce the problem

```ruby
return 'The ntp::mac_os_x_client recipe only supports macOS' unless platform_family?('mac_os_x')
```

## RuboCop version

```
0.89.0 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.7.1 x86_64-darwin19)
```

**Comment by koic** [comment]
I think this behaviour is as expected.

```
% cat example.rb
return 1, 2, 3 if condition

% bundle exec rubocop example.rb --only Lint/TopLevelReturnWithArgument
(snip)

example.rb:1:1: W: Lint/TopLevelReturnWithArgument: Top level return with argument detected.
return 1, 2, 3 if condition
^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

Because Ruby 2.7 also warns when `if` / `unless` condition modifier is specified.

```console
% ruby -vwe 'return 1, 2, 3 if false'
ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-darwin17]
-e: warning: argument of top-level return is ignored
```
